source/draw/RasterBgr.ooc

```diff
@@ -32,6 +32,22 @@ BgrRasterCanvas: class extends RasterCanvas {
 		if (this target isValidIn(position x, position y))
 			this target[position x, position y] = this target[position x, position y] blend(this pen alphaAsFloat, this pen color toBgr())
 	}
+	draw: override func ~ImageSourceDestination (image: Image, source, destination: IntBox2D) {
+		bgr: RasterBgr = null
+		if (image instanceOf?(This)) {
+			bgr = image as RasterBgr
+			bgr referenceCount increase()
+		} else if (image instanceOf?(RasterImage))
+			bgr = RasterBgr convertFrom(image as RasterImage)
+		if (bgr) {
+			if (this target size == bgr size && this target stride == bgr stride && source == destination && source size == bgr size && source leftTop x == 0 && source leftTop y == 0)
```

Comment:
I know it looks scary, but we really want `memcpy` when we can avoid per pixel loops.